gnmi_server/server_test.go

```diff
@@ -230,2294 +99,149 @@ func runServer(t *testing.T, s *Server) {
-
-	fileName = "../testdata/COUNTERS:Ethernet1.txt"
-	countersEthernet1Byte, err := ioutil.ReadFile(fileName)
-	if err != nil {
-		t.Fatalf("read file %v err: %v", fileName, err)
-	}
-	// "Ethernet1": "oid:0x1000000000003",
-	mpi_counter = loadConfig(t, "COUNTERS:oid:0x1000000000003", countersEthernet1Byte)
-	loadDB(t, rclient, mpi_counter)
-
-	// "Ethernet64:0": "oid:0x1500000000092a"  : queue counter, to work as data noise
-	fileName = "../testdata/COUNTERS:oid:0x1500000000092a.txt"
-	counters92aByte, err := ioutil.ReadFile(fileName)
-	if err != nil {
-		t.Fatalf("read file %v err: %v", fileName, err)
-	}
-	mpi_counter = loadConfig(t, "COUNTERS:oid:0x1500000000092a", counters92aByte)
-	loadDB(t, rclient, mpi_counter)
-
-	// "Ethernet68:1": "oid:0x1500000000091c"  : queue counter, for COUNTERS/Ethernet68/Queue vpath test
-	fileName = "../testdata/COUNTERS:oid:0x1500000000091c.txt"
-	countersEeth68_1Byte, err := ioutil.ReadFile(fileName)
-	if err != nil {
-		t.Fatalf("read file %v err: %v", fileName, err)
-	}
-	mpi_counter = loadConfig(t, "COUNTERS:oid:0x1500000000091c", countersEeth68_1Byte)
-	loadDB(t, rclient, mpi_counter)
-
-	// "Ethernet68:3": "oid:0x1500000000091e"  : lossless queue counter, for COUNTERS/Ethernet68/Pfcwd vpath test
-	fileName = "../testdata/COUNTERS:oid:0x1500000000091e.txt"
-	countersEeth68_3Byte, err := ioutil.ReadFile(fileName)
-	if err != nil {
-		t.Fatalf("read file %v err: %v", fileName, err)
-	}
-	mpi_counter = loadConfig(t, "COUNTERS:oid:0x1500000000091e", countersEeth68_3Byte)
-	loadDB(t, rclient, mpi_counter)
-
-	// "Ethernet68:4": "oid:0x1500000000091f"  : lossless queue counter, for COUNTERS/Ethernet68/Pfcwd vpath test
-	fileName = "../testdata/COUNTERS:oid:0x1500000000091f.txt"
-	countersEeth68_4Byte, err := ioutil.ReadFile(fileName)
-	if err != nil {
-		t.Fatalf("read file %v err: %v", fileName, err)
-	}
-	mpi_counter = loadConfig(t, "COUNTERS:oid:0x1500000000091f", countersEeth68_4Byte)
-	loadDB(t, rclient, mpi_counter)
-
-	// Load CONFIG_DB for alias translation
-	prepareConfigDb(t, namespace)
-
-	//Load STATE_DB to test non V2R dataset
-	prepareStateDb(t, namespace)
-}
+	defer mock2.Reset()
+	mockCode := 
+`
+print('No Yang validation for test mode...')
+print('%s')
+`
+	mock3 := gomonkey.ApplyGlobalVar(&sdc.PyCodeForYang, mockCode)
+	defer mock3.Reset()
+
+	s := createServer(t, 8080)
+	go runServer(t, s)
+	defer s.s.Stop()
 
-func prepareDbTranslib(t *testing.T) {
-	rclient := getRedisClient(t, sdcfg.GetDbDefaultNamespace())
-	rclient.FlushDB()
-	rclient.Close()
+	path, _ := os.Getwd()
+	path = filepath.Dir(path)
 
-	//Enable keysapce notification
-	os.Setenv("PATH", "/usr/bin:/sbin:/bin:/usr/local/bin")
-	cmd := exec.Command("redis-cli", "config", "set", "notify-keyspace-events", "KEA")
-	_, err := cmd.Output()
-	if err != nil {
-		t.Fatal("failed to enable redis keyspace notification ", err)
+	var cmd *exec.Cmd
+	cmd = exec.Command("bash", "-c", "cd "+path+" && "+"pytest -m noauth")
```

Comment:
>pytest [](http://example.com/codeflow?start=53&length=6)

pytest is for python only?

Reply:
Because pytest is more flexible than go test, I use pytest to run test with gnmi client and gnmi server.